### PR TITLE
chore: fix 40 trivial "unused variables" lint violations

### PR DESCRIPTION
**Pull request by EvanHahn**
This fixes several violations of [@typescript-eslint/no-unused-vars][0]. There are still several that I didn't feel comfortable changing, but these ones were easy to fix.

[0]: https://typescript-eslint.io/rules/no-unused-vars/

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,6 +8,12 @@ module.exports = {
       files: ['*.ts', '*.tsx'],
       rules: {
         '@typescript-eslint/no-shadow': ['error'],
+        '@typescript-eslint/no-unused-vars': [
+          'error',
+          {
+            ignoreRestSiblings: true,
```
Comment (EvanHahn) [inline .eslintrc.js:14]: This means that this is no longer a violation:

```javascript
// `foo` is never used; we only care about `rest`
const { foo, ...rest } = obj
return rest
```

File: src/frontend/hooks/useSyncState.ts [hunk 2]
```diff
@@ -56,8 +56,8 @@ function createSyncState(project: MapeoProjectApi) {
   const listeners = new Set<() => void>();
   let error: Error | undefined;
 
-  function onSyncState(state: SyncState) {
-    state = state;
+  function onSyncState(newState: SyncState) {
+    state = newState;
```
Comment (achou11) [inline src/frontend/hooks/useSyncState.ts:60]: this is more related to this rule, right? https://eslint.org/docs/latest/rules/no-shadow

not a big deal (we should eventually look into disabling that rule too)

**Review by achou11 — APPROVED** [review]
Thanks for all of these fixes! There are cases where I like to prefix the unused variable with an underscore (seems like ESLint wouldn't complain about it), but not adamant about trying to figure out the convention right now

File: src/frontend/hooks/useSyncState.ts (hunk 2 above)
Comment (EvanHahn) [reply src/frontend/hooks/useSyncState.ts:60]: If I recall, it was violating both issues. (Not 100% sure, let me know if you want me to go back and check.)

**Pull request merged**